core/validatorapi/grpc/validator.go

```diff
@@ -0,0 +1,512 @@
+	ctx := server.Context()
+	first, _ := s.GetDuties(ctx, req)
+	_ = server.Send(first)
+
+	s.subscribeEpoch(func(epoch eth2p0.Epoch) {
+		req.Epoch = pb.Epoch(epoch)
+		duties, _ := s.GetDuties(ctx, req)
+		_ = server.Send(duties)
+	})
+
+	return nil
+}
+
+// DomainData 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) DomainData(ctx context.Context, req *prysmpb.DomainRequest) (*prysmpb.DomainResponse, error) {
+	var domainType eth2p0.DomainType
+	copy(domainType[:], req.Domain)
+
+	resp, _ := s.eth2Cl.Domain(ctx, domainType, eth2p0.Epoch(req.Epoch))
+
+	return &prysmpb.DomainResponse{SignatureDomain: resp[:]}, nil
+}
+
+// DomainData 🎉 can proxy to valiatgorAPI (copy simple scheduler logic)
+func (s *BeaconNodeValidatorServer) WaitForChainStart(e *empty.Empty, server prysmpb.BeaconNodeValidator_WaitForChainStartServer) error {
+	scheduler.WaitChainStart(server.Context(), s.eth2Cl, clockwork.NewRealClock())
+	return nil
+}
+
+// WaitForActivation 🕰 requires scheduler epoch events, 🎉 but can then just query validatorAPI.
+func (s *BeaconNodeValidatorServer) WaitForActivation(req *prysmpb.ValidatorActivationRequest, server prysmpb.BeaconNodeValidator_WaitForActivationServer) error {
+	done := make(chan struct{})
+	s.subscribeEpoch(func(epoch eth2p0.Epoch) {
+		vals, _ := s.eth2Cl.ValidatorsByPubKey(server.Context(), "head", toGroupPubkeys(req.PublicKeys))
+		for _, val := range vals {
+			if val.Status == eth2v1.ValidatorStateActiveOngoing {
+				close(done)
+			}
+		}
+	})
+
+	select {
+	case <-server.Context().Done():
+		return server.Context().Err()
+	case <-done:
+		return nil
+	}
+}
+
+// ValidatorIndex 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) ValidatorIndex(ctx context.Context, req *prysmpb.ValidatorIndexRequest) (*prysmpb.ValidatorIndexResponse, error) {
+	resp, _ := s.valsProvider.ValidatorsByPubKey(ctx, "head", toGroupPubkeys([][]byte{req.PublicKey}))
+
+	return &prysmpb.ValidatorIndexResponse{Index: pb.ValidatorIndex(resp[0].Index)}, nil
+}
+
+// ValidatorStatus 🎉 can proxy to valiatgorAPI, 🙈 but will not return some data either not used in prysm or just used for logging.
+func (s *BeaconNodeValidatorServer) ValidatorStatus(ctx context.Context, req *prysmpb.ValidatorStatusRequest) (*prysmpb.ValidatorStatusResponse, error) {
+	resp, _ := s.valsProvider.ValidatorsByPubKey(ctx, "head", toGroupPubkeys([][]byte{req.PublicKey}))
+	return &prysmpb.ValidatorStatusResponse{
+		Status:                    prysmpb.ValidatorStatus(resp[0].Status),
+		Eth1DepositBlockNumber:    0, // TODO(corver): We do not have access to this, but it seems unused.
+		DepositInclusionSlot:      0, // TODO(corver): We do not have access to this, but it seems unused.
+		ActivationEpoch:           pb.Epoch(resp[0].Validator.ActivationEpoch),
+		PositionInActivationQueue: 0, // TODO(corver): We do not have access to this, it is only used for logging.
```

Comment:
this data can be queried from an execution node which is what a beacon node does, since we are connected to beacon node and not connected to execution node, we cannot get this data.